packages/nx-monorepo/src/nx-project.ts

```diff
@@ -0,0 +1,69 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+import * as path from "path";
+import { Component, JsonFile, Project } from "projen";
+import { NodeProject } from "projen/lib/javascript";
+import { buildExecutableCommand } from "./nx-monorepo";
+
+export class NxProject extends Component {
+  static of(project: Project): NxProject {
+    return project.components.find((c) => c instanceof NxProject) as NxProject;
+  }
+
+  private readonly implicitDependencies: string[] = [];
+  constructor(project: Project) {
+    super(project);
+  }
+
+  public addImplicitDependency(dependee: Project | string) {
+    this.implicitDependencies.push(
+      dependee instanceof Project ? dependee.name : dependee
+    );
+  }
+
+  synthesize() {
+    if (this.project instanceof NodeProject) {
+      this.project
+        .tryFindObjectFile("package.json")
+        ?.addOverride("nx.implicitDependencies", this.implicitDependencies);
```

Comment:
Should this use [addToArray](https://github.com/projen/projen/blob/main/src/object-file.ts#L156), which would support existing values of `nx.implicitDependencies` being added prior to this call.

Reply:
Would prefer to keep as is as otherwise I will need NodeProject vs Non-NodeProject branching in two places opposed to one.